libbeat/template/processor_test.go

```diff
@@ -356,35 +389,72 @@ func TestDynamicTemplate(t *testing.T) {
 				Type: "object", ObjectType: "scaled_float",
 				Name: "core.*.pct", ScalingFactor: 100, ObjectTypeMappingType: "float",
 			},
-			expected: common.MapStr{
-				"core.*.pct": common.MapStr{
-					"mapping": common.MapStr{
-						"type":           "scaled_float",
-						"scaling_factor": 100,
+			expected: []common.MapStr{
+				common.MapStr{
+					"core.*.pct": common.MapStr{
+						"mapping": common.MapStr{
+							"type":           "scaled_float",
+							"scaling_factor": 100,
+						},
+						"match_mapping_type": "float",
+						"path_match":         "core.*.pct",
+					},
+				},
+			},
+		},
+		{
+			field: common.Field{
+				Type: "object", ObjectTypeParams: []common.ObjectTypeCfg{
+					{ObjectType: "float", ObjectTypeMappingType: "float"},
+					{ObjectType: "boolean"},
+					{ObjectType: "scaled_float", ScalingFactor: 10000},
+				},
+				Name: "context",
+			},
+			expected: []common.MapStr{
+				common.MapStr{
+					"context": common.MapStr{
+						"mapping":            common.MapStr{"type": "float"},
+						"match_mapping_type": "float",
+						"path_match":         "context.*",
+					},
+				},
+				common.MapStr{
+					"context": common.MapStr{
+						"mapping":            common.MapStr{"type": "boolean"},
+						"match_mapping_type": "boolean",
+						"path_match":         "context.*",
+					},
+				},
+				common.MapStr{
+					"context": common.MapStr{
+						"mapping":            common.MapStr{"type": "scaled_float", "scaling_factor": 10000},
+						"match_mapping_type": "*",
+						"path_match":         "context.*",
 					},
-					"match_mapping_type": "float",
-					"path_match":         "core.*.pct",
 				},
 			},
 		},
 	}
 
-	for _, numericType := range []string{"byte", "double", "float", "long", "short"} {
+	for _, numericType := range []string{"byte", "double", "float", "long", "short", "boolean"} {
```

Comment:
Why adding boolean here?

Reply:
Because I added it here https://github.com/elastic/beats/pull/9772/files/426be2defdc18b60534cbaaaadd075667ee3cdf1#diff-8de3d52ef44cee7af5f205820f2d55e9R302.

When using `scaled_float` as `object_type`, an `*` is used as `match_type`. Therefore every other allowed `object_type` needs to be declared separately.

Reply:
Thanks for the clarification :+1: